src/pages/Home.tsx

```diff
@@ -1,11 +1,184 @@
+import Lead from "../service/models/lead.model";
+import Need from "../service/models/need.model";
+import CreateOrEditNeedModal from "../components/needs-and-leads/CreateOrEditNeedModal";
+import CreateOrEditLeadModal from "../components/needs-and-leads/CreateOrEditLeadModal";
+import { MUTATE_NEED } from "../service/apollo/mutations";
+
+import heartImage from "../images/homepage_heart.jpg";
+import maskImage from "../images/homepage_mask.jpg";
+import threePeopleImage from "../images/homepage_three_people.jpg";
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: "100%"
+  },
+  head: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    background: `url(${heartImage}) no-repeat center center`,
+    backgroundSize: "cover",
+    height: "50vh",
+    width: "100%",
+    minHeight: "260px"
+  },
+  container: {
+    display: "flex",
+    paddingTop: "25px",
+    flexWrap: "wrap"
+  },
+  submitNeed: {
+    width: "45vw",
+    height: "22vh",
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    textAlign: "center",
+    minWidth: "210px",
+    minHeight: "185px",
+    marginTop: "10px",
+    background: `linear-gradient(rgba(255, 255, 255, .7), rgba(255, 255, 255, .7)), url(${threePeopleImage}) no-repeat center center `,
+    backgroundSize: "cover",
+    color: "#5b5b5b",
+    fontSize: "17px"
+  },
+  submitLead: {
+    width: "45vw",
+    height: "22vh",
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    textAlign: "center",
+    minWidth: "210px",
+    minHeight: "185px",
+    marginTop: "10px",
+    background: `linear-gradient(rgba(255, 255, 255, .7), rgba(255, 255, 255, .7)), url(${maskImage}) no-repeat center center `,
+    backgroundSize: "cover",
+    color: "#5b5b5b",
+    fontSize: "17px"
+  },
+  text: {
+    marginBottom: "15px",
+    fontWeight: 600
+  },
+  textHead: {
+    color: "white",
+    textAlign: "center",
+    width: "30vw",
+    fontWeight: 500,
+    fontSize: "16px",
+    lineHeight: "26px",
+    minWidth: "140px"
+  },
+  submit: {
+    textAlign: "center",
+    bottom: theme.spacing(5)
+  },
+  button: {
+    fontWeight: 400
+  },
+  title: {
+    marginTop: theme.spacing(2),
+    marginLeft: theme.spacing(3),
+    marginBottom: theme.spacing(1)
+  },
+  searchTool: {
+    marginBottom: theme.spacing(1)
+  },
+  modal: {
+    outline: "none"
+  }
+}));
 
 function Home() {
+  const styles = useStyles();
+  const [showNeedSubmitModal, setShowNeedSubmitModal] = useState(false);
+  const [showLeadSubmitModal, setShowLeadSubmitModal] = useState(false);
+  const [error, setError] = useState("");
+  const [mutateNeed] = useMutation(MUTATE_NEED);
+
+  const onNeedSubmit = (need: Need) => {
+    mutateNeed({
+      variables: { request: need }
+    })
+      .then(() => setShowNeedSubmitModal(false))
+      .catch(() => setError("Internal server error, please try again later"));
```

Comment:
catch the server error and display it.

Reply:
Done, I copied from Signup.tsx: const errorMsg = error?.response?.data?.error;